### PR TITLE
refactor(allowance): get action

### PR DESCRIPTION
**Pull request by im-adithya**
### Describe the changes you have made in this PR

Switching allowance/delete to TS

### Link this PR to an issue

Relates to #989

### Type of change (Remove other not matching type)

- `refactor`: Migration to TypeScript

### How has this been tested?

Should add tests, would be done after review

### Checklist

- [x] My code follows the style guidelines of this project and performed a self-review of my own code
- [x] New and existing tests pass locally with my changes
- [x] I checked if I need to make corresponding changes to the documentation (and made those changes if needed)

### CODE REVIEW

**Review by escapedcat — DISMISSED** [review]
🚀

File: src/extension/background-script/db.ts [hunk 1]
```diff
@@ -1,11 +1,11 @@
 import Dexie from "dexie";
 import "fake-indexeddb/auto";
 import browser from "webextension-polyfill";
-import type { Allowance, Payment, Blocklist } from "~/types";
+import type { Allowance, Transaction, Blocklist } from "~/types";
 
 class DB extends Dexie {
   allowances: Dexie.Table<Allowance, number>;
-  payments: Dexie.Table<Payment, number>;
+  payments: Dexie.Table<Transaction, number>;
```
Comment (bumi) [inline src/extension/background-script/db.ts:8]: Why that? I don't think this is correct The data in the table does not match all the attributes of `Transaction`

@escapedcat ?

Comment (escapedcat) [reply src/extension/background-script/db.ts:8]: Good point. I thought this has been done for a reason but should have asked. @im-adithya can you elaborate?

Comment (im-adithya) [reply src/extension/background-script/db.ts:8]: I was getting a TS error as we have the `payments` type set to `Transaction[]` in types:

https://github.com/getAlby/lightning-browser-extension/blob/84a2d8fe8b77cf4a7170d65b09bec163eb2d6246/src/extension/background-script/actions/allowances/get.js#L22


https://github.com/getAlby/lightning-browser-extension/blob/84a2d8fe8b77cf4a7170d65b09bec163eb2d6246/src/types.ts#L325

Comment (im-adithya) [reply src/extension/background-script/db.ts:8]: And we have this component saying transactions specifically and also uses Transaction type, so I thought changing the payments type to Payment[] would be breaking things:
https://github.com/getAlby/lightning-browser-extension/blob/84a2d8fe8b77cf4a7170d65b09bec163eb2d6246/src/app/screens/Home/index.tsx#L197

Comment (escapedcat) [reply src/extension/background-script/db.ts:8]: Thanks!  
I think the payments in the db and the payments attached to an allowance should have the type `payment`.  
Currently only the transaction-table is using `transaction` I think.  
Following does that path I [created a new PR](https://github.com/im-adithya/lightning-browser-extension/pull/2) which to extend this PR. Currently there are a lot of changes, but once this one is rebased with our master it should only have one commit left.
I rebased this PR but I can't force push onto this one due to missing permissions. Not sure why. I believe maintainers usually can push into contributor-forks 🤷

**Comment by escapedcat** [comment]
- Typecheck is failing
- Is this needed:
  ```ts
  totalAmount: number;
  ```
  hav ewe stored `totalAmount` in the index-db before? If a user has strings in there, will this break?

**Review by escapedcat — APPROVED** [review]

**Pull request merged**